<?xml version="1.0"?>
<robot name="bot">

    <material name="blue">
        <color rgba="0.0 0.0 0.8 1.0"/>
    </material>

    <material name="green">
        <color rgba="0.0 0.8 0.0 1.0"/>
    </material>

    <material name="grey">
        <color rgba="0.2 0.2 0.2 1.0"/>
    </material>

    <material name="red">
        <color rgba="0.8 0.0 0.0 1.0"/>
    </material>

    <material name="white">
        <color rgba="1.0 1.0 1.0 1.0"/>
    </material>

    <link name="base_link">
        <visual>
            <geometry>
                <cylinder length="0.6" radius="0.2"/>
            </geometry>
            <origin xyz="0 0 0.3" rpy="0 0 0"/>
            <material name="white"/>
        </visual>
    </link>

    <link name="base_2">
        <visual>
            <geometry>
                <cylinder length="0.48" radius="0.16"/>
            </geometry>
            <origin xyz="0 0 0.04" rpy="0 0 0"/>
            <material name="red"/>
        </visual>
    </link>

    <joint name="base_link_to_base_2_linear" type="continuous">
        <parent link="base_link"/>
        <child link="base_2"/>
        <origin xyz="0 0 0.6"/>

        <axis xyz="0 0 1"/>
    </joint>

    <link name="base_head">
        <visual>
            <geometry>
                <cylinder length="0.5" radius="0.16"/>
            </geometry>
            <origin rpy="0 1.57 0" xyz="0.04 0 0"/>
            <material name="white"/>
        </visual>
    </link>

    <joint name="base_2_to_base_head" type="fixed">
        <parent link="base_2"/>
        <child link="base_head"/>
        <origin xyz="0 0 0.28"/>
    </joint>

    <link name="upper_arm">
        <visual>
            <geometry>
                <box size="0.06 1.2 0.17"/>
            </geometry>
            <origin xyz="0 0.525 0"/>
            <material name="blue"/>
        </visual>
    </link>

    <joint name="base_head_to_upper_arm" type="revolute">
        <parent link="base_head"/>
        <child link="upper_arm"/>
        <origin xyz="0.32 0 0" rpy="0 0 3.14"/>

        <limit effort="1000" upper="1.22" lower="0" velocity="1"/>
        <axis xyz="1 0 0"/>
    </joint>

    <link name="lower_arm">
        <visual>
            <geometry>
                <box size="0.03 0.8 0.11"/>
            </geometry>
            <origin xyz="0.03 0.325 0"/>
            <material name="green"/>
        </visual>
    </link>

    <joint name="upper_arm_to_lower_arm" type="revolute">
        <parent link="upper_arm"/>
        <child link="lower_arm"/>
        <origin xyz="0 1.05 0"/>

        <limit effort="1000" upper="1.05" lower="-2.095" velocity="1"/>
        <axis xyz="0.03 0 0"/>
    </joint>

</robot>